### PR TITLE
doc: add comparison with other tools

### PR DESCRIPTION
**Pull request by VoVAllen**
Signed-off-by: Jinjing.Zhou <allenzhou@tensorchord.ai>

### CODE REVIEW

File: docs/tips/comparison.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Envd vs. Other tools
```
Comment (kemingy) [inline docs/tips/comparison.md:1]: Should all `envd` be in code format and small case?

File: docs/tips/comparison.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## envd Benefits
+Benefits of using envd include:
+- Clear dev environment setup and management for each project. No longer need to worry about the conflict of CUDA, system package, virtualenv, and so on. Each project will have its own dev environment. And you can easily use the environment on any machines, and share it with your collaborators. 
+- Unify dev and production environments by envd's multi-target supports.
+- Integrate with container ecosystem. envd can be used in any container, such as Docker, Kubernetes, and so on.
+- Dataset support. envd can be used to download and preprocess datasets.
+
+## How does it work?
+- Parallize the image build process. Powered by buildkit, envd can parallize different tools installation to minimize the build time.
+- Cache support for common installation tools, such as pip, conda, apt, and so on. envd had builtin cache for these tools, which minimize the time to wait when you make modifications to your environment.
```
Comment (kemingy) [inline docs/tips/comparison.md:17]: I think we can emphasize how bulidkit cache is better than the normal docker build cache.

File: docs/.vitepress/config/sidebar.ts [hunk 3]
```diff
@@ -85,6 +85,7 @@ export const sidebar: DefaultTheme.Sidebar = {
       text: 'Tips and Tricks',
       collapsible: true,
       items:[
+        { text: 'Envd vs. other software', link:'/tips/comparison' },
```
Comment (gaocegege) [inline docs/.vitepress/config/sidebar.ts:88]: Could we place it in the last? After https://github.com/tensorchord/envd-docs/pull/90/files#diff-28b8d9c36ba8331d833f98071811f63570f78ce49c485f289c73aa5417aa931dL92

File: docs/tips/comparison.md (hunk 2 above)
Comment (VoVAllen) [reply docs/tips/comparison.md]: Added a new sentence to describe normal docker build experience

File: docs/tips/comparison.md (hunk 1 above)
Comment (kemingy) [reply docs/tips/comparison.md]: To align with other pages, "envd" should be in the code format, quoted by "`". Can check the source code of other pages.

Comment (VoVAllen) [reply docs/tips/comparison.md]: Added "`"

File: docs/tips/comparison.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- Clear dev environment setup and management for each project. No longer need to worry about the conflict of CUDA, system package, virtualenv, and so on. Each project will have its own dev environment. And you can easily use the environment on any machines, and share it with your collaborators. 
+- Unify dev and production environments by `envd`'s multi-target supports.
+- Integrate with container ecosystem. `envd` can be used in any container, such as Docker, Kubernetes, and so on.
+- Dataset support. `envd` can be used to download and preprocess datasets.
+
+## How does it work?
+- Parallize the image build process. Powered by buildkit, `envd` can parallize different tools installation to minimize the build time.
+- Cache support for common installation tools, such as pip, conda, apt, and so on. In normal dockerfile build, each layer depends on the previous layer. Andy change happens before will trigger the re-build for all the later stages, which increase the build time a lot. `envd` had builtin cache support for most common-used tools and reduce the dependency between them, which minimize the time to wait when you make modifications to your environment.
```
Comment (kemingy) [inline docs/tips/comparison.md:17]:
```suggestion
- Cache support for common installation tools, such as pip, conda, apt, and so on. In normal dockerfile build, each layer depends on the previous layer. Any change happens before will trigger the re-build for all the later stages, which increases the build time a lot. `envd` had builtin cache support for most common-used tools and reduce the dependency between them, which minimize the time to wait when you make modifications to your environment. 
```

File: docs/.vitepress/config/sidebar.ts [hunk 5]
```diff
@@ -100,7 +100,8 @@ export const sidebar: DefaultTheme.Sidebar = {
         { text: 'Per-user Config', link:'/tips/config'},
         { text: 'Remote Cache', link: '/tips/cache'},
         { text: 'CI/CD Integration', link:'/tips/ci'},
-        { text: 'Multi-target Build', link:'/tips/multitarget'}
+        { text: 'Multi-target Build', link:'/tips/multitarget'},
+        { text: 'envd vs. other software', link:'/tips/comparison' }
```
Comment (gaocegege) [inline docs/.vitepress/config/sidebar.ts:104]: Please copy the file to zh and update https://github.com/tensorchord/envd-docs/pull/90/files#diff-28b8d9c36ba8331d833f98071811f63570f78ce49c485f289c73aa5417aa931dR44

**Comment by gaocegege** [comment]
Please @ me when it is ready to review.

**Comment by VoVAllen** [comment]
@gaocegege done

File: docs/tips/comparison.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+- `envd` can export container used for production or pipeline stages, which narrow down the gap between development and production. You can build multiple environments for different purposes(research, development, serving, data processing, etc.) from the same file.
+<!-- We can add data related section once finished -->
+
+
+## `envd` vs. docker
+`envd` is build on top of the docker ecosystem, and fully compatible with docker.
+- `envd` parallize the image build process, which accelerates the build process. 
+- docker uses dockerfile as the major language. `envd` uses starlark as the language, which is more familiar to the data science workers.
+- `envd` supports multi-target build from single file, while docker usually need to write multiple files, which makes reuse of the building block difficult.
```
Comment (terrytangyuan) [inline docs/tips/comparison.md:32]: Docker supports multi-stage build: https://docs.docker.com/develop/develop-images/multistage-build/

Is it the same as the multi-target build you mentioned here?

File: docs/tips/comparison.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+`envd` doesn't conflict with conda. You can also use conda inside `envd` by `install.conda_packages`
+- `envd` environment is built from the pure official image, there will be no legacy installation artifacts. Only those you declare in the file will appear in the environment. conda's environment isolation functionality depends on environment variable and rewrite library's rpath. User might meet conflict between system package and conda package (such as CUDA). In corner cases, it might results in unexpected behaviors.
+- `envd` can export container used for production or pipeline stages, which narrow down the gap between development and production. You can build multiple environments for different purposes(research, development, serving, data processing, etc.) from the same file.
+<!-- We can add data related section once finished -->
+
+
+## `envd` vs. docker
+`envd` is build on top of the docker ecosystem, and fully compatible with docker.
+- `envd` parallize the image build process, which accelerates the build process. 
+- docker uses dockerfile as the major language. `envd` uses starlark as the language, which is more familiar to the data science workers.
```
Comment (terrytangyuan) [inline docs/tips/comparison.md:31]: workers -> practitioners

File: docs/tips/comparison.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+## `envd` vs. conda
+`envd` doesn't conflict with conda. You can also use conda inside `envd` by `install.conda_packages`
+- `envd` environment is built from the pure official image, there will be no legacy installation artifacts. Only those you declare in the file will appear in the environment. conda's environment isolation functionality depends on environment variable and rewrite library's rpath. User might meet conflict between system package and conda package (such as CUDA). In corner cases, it might results in unexpected behaviors.
+- `envd` can export container used for production or pipeline stages, which narrow down the gap between development and production. You can build multiple environments for different purposes(research, development, serving, data processing, etc.) from the same file.
+<!-- We can add data related section once finished -->
+
+
+## `envd` vs. docker
+`envd` is build on top of the docker ecosystem, and fully compatible with docker.
```
Comment (terrytangyuan) [inline docs/tips/comparison.md:29]: build -> built

File: docs/tips/comparison.md (hunk 6 above)
Comment (VoVAllen) [reply docs/tips/comparison.md]: It's not multi-stage here I think. Multi-target means you can define the docker image for development/training job/serving within a single file and reuse the building block across them.

**Comment by VoVAllen** [comment]
Thanks for @terrytangyuan's suggestion! Made a new pass with Grammarly's help.

cc @gaocegege

**Review by gaocegege — APPROVED** [review]

**Comment by gaocegege** [comment]
Thanks for your contribution! :tada: :+1:

**Pull request merged**